### PR TITLE
Analyze all Dart code; fix existing warnings

### LINKED ISSUE #1392: Fix Dart analyzer errors/warnings/hints in web directories
The dartanalyzer script has been missing these. There are a number of things to cleanup.

### PR DESCRIPTION
**Pull request by yjbanov**
Our analyzer was missing a bunch of directories. This PR fixes that. We now analyze everything.

Fixes #1392 and #1395

### CODE REVIEW

File: modules/angular2/src/facade/collection.es6 [hunk 1]
```diff
@@ -193,6 +193,9 @@ export class ListWrapper {
   static slice(l:List, from:int, to:int):List {
     return l.slice(from, to);
   }
+  static splice(l:List, from:int, to:int):List {
+    return l.splice(from, to);
```
Comment (vicb) [inline modules/angular2/src/facade/collection.es6]: JS `splice` is `array.splice(start, deleteCount[, item1[, item2[, ...]]])`

We should keep the same semantics:
- `start` could be < 0, it should be supported in Dart (it is not in this PR, see `fill()`, we should factorize code)
- `to` should be `deleteCount` - there is a bug in this JS impl,
- we should be able to add items as with vanilla JS

**Comment by kevmoo** [comment]
:metal:

**Comment by kevmoo** [comment]
Update the library find logic, too - https://github.com/angular/angular/pull/1411/files#diff-f35e8ec5a6ad6827577b472fe175be26R37

At the moment, this only gets files directly under `lib/`. We need to look for every Dart file.

**Comment by yjbanov** [comment]
You are free to look at the code, of course, but the PR is still WIP, as indicated by the label ;)

@vicb, good catch!

**Comment by kevmoo** [comment]
Change...

```dart
 var sources = [].slice.call(glob.sync('lib/*.dart', {
 cwd: dirName
}));
```

to

```dart
 var sources = [].slice.call(glob.sync('lib/**/*.dart', {
 cwd: dirName
}));
```

File: modules/angular2/src/facade/collection.es6 (hunk 1 above)
Comment (yjbanov) [inline modules/angular2/src/facade/collection.es6]: I'll fix the `deleteCount` issue but I'm not going for full semantics match here or match the functionality. In fact, I don't like how `fill()` is implemented at all. It nullifies Dart's type inference. In Dart `new List.filled(len, 0)` infers as "list of integers none of which are null". However, `ListWrapper.fill` infers as "list of whatever".

**Comment by yjbanov** [comment]
Removed WIP label. This is ready for review now.

File: modules/angular2/test/facade/collection_spec.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        expectList(ListWrapper.splice(list, 1, 3), [2, 3, 4]);
+        expectList(list, [1, 5, 6]);
+      });
+    });
+ });
+}
+
+// TODO(vsavkin): for some reason expect(list1).toBe(list2) does not work
+function expectList(l1:List, l2:List):void {
+  expect(ListWrapper.equals(l1, l2)).toBe(true);
```
Comment (vsavkin) [inline modules/angular2/test/facade/collection_spec.js]: use toEqual

File: modules/angular2_material/src/components/switcher/switch.js [hunk 3]
```diff
@@ -61,7 +65,7 @@ export class MdSwitch {
   }
 
   onKeydown(event: KeyboardEvent) {
-    if (event.keyCode == KeyCodes.SPACE) {
+    if (event.keyCode == KEY_SPACE) {
```
Comment (vsavkin) [inline modules/angular2_material/src/components/switcher/switch.js]: ===

Comment (yjbanov) [inline modules/angular2_material/src/components/switcher/switch.js]: done

File: modules/angular2/test/facade/collection_spec.js (hunk 2 above)
Comment (yjbanov) [inline modules/angular2/test/facade/collection_spec.js]: done

**Pull request merged**